clang/test/SemaSYCL/fake-accessors.cpp

```diff
@@ -11,46 +13,44 @@ class accessor {
 };
 } // namespace sycl
 } // namespace cl
-} // namespace foo
+} // namespace fake
 
 class accessor {
 public:
   int field;
 };
 
-typedef cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write,
-                           cl::sycl::access::target::global_buffer>
-    MyAccessorTD;
-
-using MyAccessorA = cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write,
-                                       cl::sycl::access::target::global_buffer>;
-
-template <typename name, typename Func>
-__attribute__((sycl_kernel)) void kernel(const Func &kernelFunc) {
-  kernelFunc();
-}
-
 int main() {
-  foo::cl::sycl::accessor acc = {1};
+  fake::cl::sycl::accessor FakeAccessor = {1};
   accessor acc1 = {1};
 
-  cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write> accessorA;
-  cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write> accessorB;
-  cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write> accessorC;
-    kernel<class fake_accessors>(
-        [=]() {
-          accessorA.use((void*)(acc.field + acc1.field));
+  sycl::accessor<int, 1, sycl::access::mode::read_write> accessorA;
+  sycl::accessor<int, 1, sycl::access::mode::read_write> accessorB;
+  sycl::accessor<int, 1, sycl::access::mode::read_write> accessorC;
+
```

Comment:
Not sure what this test is trying to accomplish. The test seems to pass the 3 accessors with just different kernel names.
I will look at the history/git blame of this file to see if that helps.

Reply:
From the kernel names it looks like the original intention was to check the typedef and alias declared above. Not sure when/why that changed.

Reply:
I recall something about this test. It was trying to check that our detection of SYCL accessors is safe enough, so the compiler doesn't process random class named "accessor" as SYCL accessor and it doesn't miss real SYCL accessors which were aliased or typedef-ed. Although, I don't know why `MyAccessorTD` and `MyAccessorA` typedef and alias are not used anymore, IMO they should be.

Reply:
Did the history give you an idea of what happened? I assume  `MyAccessorTD `and `MyAccessorA ` was removed by accident at some point. If so, can you update this test?

Reply:
I haven't had a chance to resolve this yet, which is why I left it open. I will be updating it by EOW.